.github/workflows/main.yml

```diff
@@ -0,0 +1,76 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+  
+   # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+      
+      # Install Node dependency
+      - uses: actions/setup-node@v2.1.5  
+      - run: npm install
+      
+      # Get version
+      - name: get-npm-version
+        id: package-version
+        uses: martinbeentjes/npm-get-version-action@master
+      - run: echo "${{steps.package-version.outputs.current-version}}" 
+     
+     # Package the plugin and upload artifact
+      - run: npm pack
+      - run: mkdir -p output/artifact
+      - run: cp *.tgz output/artifact/
+      - uses: actions/upload-artifact@v2
```

Comment:
should we also configure a retain period? I don't think we are interested in keeping old artifacts for longer than couple of months.

Reply:
Added changes for retention period, cache and tag